README.md

```diff
@@ -4,221 +4,501 @@
-client = v3f.Client('framesd:8081', user='user1', password='pass')
+client = v3f.Client("framesd:8081", user="iguazio", password="mypass", container="users")
 ```
-Note: When running from within the managed jupyter notebook on the iguazio platform there is no need to add credentials as this is handled by the platform.  
-Next, for every operation we need to provide a `backend`, and a `table` parameters and optionally other function specific arguments.
 
-### Create
-Creates a new table for the desired backend. Not all backends require a table to be created prior to ingestion. For example KV table will be created while ingesting new data, on the other hand since TSDB tables have mandatory fields we need to create a table before ingesting new data.  
+<a id="client-common-method-params"></a>
+### Common Client Method Parameters
+
+All client methods receive the following common parameters:
+
+- <a id="client-method-param-backend"></a>**backend** &mdash; the backend data type for the operation.
+  See the backend-types descriptions in the [overview](#backend-types).
+
+  - **Type:** String
+  - **Valid Values:** `"csv"` | `"kv"` | `"stream"` | `"tsdb"`
+  - **Requirement:** Required
+
+- <a id="client-method-param-table"></a>**table** &mdash; the relative path to the backend data &mdash; a directory in the target platform data container (as configured for the client object) that represents a platform data collection, such as a TSDB or NoSQL table or a stream.
+  For example, `"mytable"` or `"examples/tsdb/my_metrics"`.
+
+  - **Type:** String
+  - **Requirement:** Required
+
+Additional method-specific parameters are described for each method.
+
+<a id="method-create"></a>
+### create Method
+
+Creates a new data collection (table/stream) in a platform data container according to the configured backend.
+
+The `create` method is supported by the `tsdb` and `stream` backends, but not by the `kv` backend, because NoSQL tables in the platform don't need to be created prior to ingestion; when ingesting data into a table that doesn't exist, the table is automatically created.
+
+- [Syntax](#method-create-syntax)
+- [`tsdb` backend `create` parameters](#method-create-params-tsdb)
+- [`stream` backend `create` parameters](#method-create-params-stream)
+
+<a id="method-create-syntax"></a>
+#### Syntax
+
 ```python
-client.create(backend=<backend>, table=<table>, attrs=<backend_specefic_attributes>)
+create(backend=<backend>, table=<table or stream>, attrs=<backend-specific parameters>)
 ```
 
-#### backend specific parameters
-##### TSDB
-* rate 
-* aggregates (optional)
-* aggregation-granularity (optional)
+<a id="method-create-params-tsdb"></a>
+#### tsdb Backend create Parameters
+
+- <a id="method-create-tsdb-param-rate"></a>**rate** (Required) &mdash; `string` &mdash; the ingestion rate TSDB's metric-samples, as `"[0-9]+/[smh]"` (where `s` = seconds, `m` = minutes, and `h` = hours); for example, `"1/s"` (one sample per minute).
```

Comment:
... the approximate ingestion rate for the TSDB's metric-samples ...

Reply:
@talIguaz the next line in this parameter description is "The rate should be calculated according to the slowest expected ingestion rate.". Doesn't this cover what you wanted to add?

Reply:
Don't think `slowest` is necessarily better. i guess an average rate will suite better.
But it depends on if there's a big difference between the expected rates. if so we need to think what will be better, otherwise i guess avg should work.

Reply:
@talIguaz currently, in our doc-site TSDB CLI tutorial, Frames v3io/tutorials NB, and Frames README file we say that the TSDB ingestion rate should be set according to the slowest expected ingestion rate. This doc was added based on Golan's guidelines. @dinal please let me know whether we need to change to this guideline, in light of Tal's input, and if so, what do you want the new guideline to be.